### PR TITLE
UI: Ember upgrade: Handle deprecation router service from host

### PR DESCRIPTION
**Pull request by hellobontempo**
### Description
Handles [Use alias for inject router service from host application](https://ember-engines.com/docs/deprecations#-use-alias-for-inject-router-service-from-host-application). This PR is based on initial work done by https://github.com/hashicorp/vault/pull/28281, which I recommend comparing to the changes here.

In summary this PR: 
- replaces `router` engine dependencies with aliased `app-router`
- call the alias `@service('app-router') router` within engines instead of `@service router`

- I decided to handle the `transition-to` slightly differently than the original PR. I uninstalled [ember-router-helpers](https://github.com/rwjblue/ember-router-helpers) and replaced it with a global `transition-to` helper that uses `getOwner` to dynamically determine which router to use.
- Similarly, I refactored `LinkedBlock` (which is a core component used across engines and the main app) to lookup the appropriate router service via a getter, which felt more stable instead of attempting to inject a service that doesn't exist (which according to above, would throw an error)

### TODO only if you're a HashiCorp employee
- [ ] **Backport Labels:** If this PR is in the ENT repo and needs to be backported, backport  
  to N, N-1, and N-2, using the `backport/ent/x.x.x+ent` labels. If this PR is in the CE repo, you should only backport to N, using the `backport/x.x.x` label, not the enterprise labels.
    - [ ] If this fixes a critical security vulnerability or [severity 1](https://www.hashicorp.com/customer-success/enterprise-support) bug, it will also need to be backported to the current [LTS versions](https://developer.hashicorp.com/vault/docs/enterprise/lts#why-is-there-a-risk-to-updating-to-a-non-lts-vault-enterprise-version) of Vault. To ensure this, use **all** available enterprise labels.
- [ ] **ENT Breakage:** If this PR either 1) removes a public function OR 2) changes the signature
  of a public function, even if that change is in a CE file, _double check_ that
  applying the patch for this PR to the ENT repo and running tests doesn't
  break any tests. Sometimes ENT only tests rely on public functions in CE
  files.
- [ ] **Jira:** If this change has an associated Jira, it's referenced either
  in the PR description, commit message, or branch name.
- [ ] **RFC:** If this change has an associated RFC, please link it in the description.
- [ ] **ENT PR:** If this change has an associated ENT PR, please link it in the
  description. Also, make sure the changelog is in this PR, _not_ in your ENT PR.

### CODE REVIEW

File: ui/lib/config-ui/addon/components/messages/page/create-and-edit.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ import { isAfter } from 'date-fns';
  */
 
 export default class MessagesList extends Component {
-  @service router;
+  @service('app-router') router;
```
Comment (hellobontempo) [inline ui/lib/config-ui/addon/components/messages/page/create-and-edit.js:26]: this is how you pass the service if the variable you assign it to (i.e. `router`) differs from the service name

File: ui/lib/core/addon/helpers/transition-to.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import Helper from '@ember/component/helper';
+import { getOwner } from '@ember/application';
+
+/*
+template helper that replaces ember-router-helpers https://github.com/rwjblue/ember-router-helpers
+example:
+<MyForm @onSave={{transition-to "vault.cluster.some.route.item" "item-id"}} />
+<MyForm @onSave={{transition-to "vault.cluster.some.external.route"  external=true}} />
+*/
+export default class TransitionTo extends Helper {
```
Comment (hellobontempo) [inline ui/lib/core/addon/helpers/transition-to.js:15]: Open to other ideas here, this seemed like a nice and simple way to dynamically use the `transition-to` helper without knowing the router ahead of time. It also was nice to remove an dependency 🎉 

For reference, Chelsea had made a specific `engine-transition-to` helper [here](https://github.com/hashicorp/vault/pull/28281/files#diff-dcaf011bd9819197ab45fe8516708c96243752d62bf3efe0233da2091edfd9b5). Which is what I used as a basis for building this component. 

Also there are no instances now where we pass `external=true` but I've copied her test to validate this works as expected.

File: ui/lib/pki/addon/components/page/pki-configure-create.ts [hunk 3]
```diff
@@ -6,8 +6,8 @@
 import Component from '@glimmer/component';
 import { service } from '@ember/service';
 import { tracked } from '@glimmer/tracking';
-import type Store from '@ember-data/store';
-import type Router from '@ember/routing/router';
+import type StoreService from '@ember-data/store';
```
Comment (hellobontempo) [inline ui/lib/pki/addon/components/page/pki-configure-create.ts:9]: I like including `Service` on the end of the name here, so added that!

File: ui/lib/core/addon/components/linked-block.js [hunk 4]
```diff
@@ -4,7 +4,7 @@
  */
 
 import Component from '@glimmer/component';
-import { service } from '@ember/service';
+import { getOwner } from '@ember/owner';
```
Comment (Monkeychip) [inline ui/lib/core/addon/components/linked-block.js:7]: 👍🏻

File: ui/lib/core/addon/helpers/transition-to.js (hunk 2 above)
Comment (Monkeychip) [reply ui/lib/core/addon/helpers/transition-to.js:15]: I like this approach a lot. Nice work!

File: ui/lib/core/addon/mixins/replication-actions.js [hunk 5]
```diff
@@ -11,7 +11,6 @@ import { task } from 'ember-concurrency';
 
 export default Mixin.create({
   store: service(),
-  router: service(),
```
Comment (Monkeychip) [inline ui/lib/core/addon/mixins/replication-actions.js:14]: I wonder why we had this service in here to begin with🤔

Comment (hellobontempo) [reply ui/lib/core/addon/mixins/replication-actions.js:14]: Yeah, I was going to move the service to the components that used the `ReplicationActions` mixin but then realized this wasn't used anywhere in the mixin or where the mixin was used?? My guess is left behind during a refactor?

Comment (hellobontempo) [reply ui/lib/core/addon/mixins/replication-actions.js:14]: But would definitely appreciate a second set of eyes double checking!

**Review by Monkeychip — APPROVED** [review]
I was able to smoke test replication and other transitions. Nice work!

**Comment by hellobontempo** [comment]
✅ enterprise tests
Screenshot 2024-10-08 at 9 01 11 AM

**Pull request merged**